stl/inc/ranges

```diff
@@ -1328,8 +1328,190 @@ namespace ranges {
+            }
+
+            constexpr _Iterator& operator++() noexcept(noexcept(++_Current)) /* strengthened */ {
+                ++_Current;
+                return *this;
+            }
+            constexpr _Iterator operator++(int) noexcept(
+                is_nothrow_copy_constructible_v<_Iterator>&& noexcept(++_Current)) /* strengthened */ {
+                auto _Tmp = *this;
+                ++*this;
+                return _Tmp;
+            }
+
+            constexpr _Iterator& operator--() noexcept(noexcept(--_Current)) /* strengthened */ {
+                --_Current;
+                return *this;
+            }
+            constexpr _Iterator operator--(int) noexcept(
+                is_nothrow_copy_constructible_v<_Iterator>&& noexcept(--_Current)) /* strengthened */ {
+                auto _Tmp = *this;
+                --*this;
+                return _Tmp;
+            }
+
+            constexpr _Iterator& operator+=(difference_type _Off) noexcept(
+                noexcept(_Current += _Off)) /* strengthened */ {
+                _Current += _Off;
+                return *this;
+            }
+            constexpr _Iterator& operator-=(difference_type _Off) noexcept(
+                noexcept(_Current -= _Off)) /* strengthened */ {
+                _Current -= _Off;
+                return *this;
+            }
+            _NODISCARD constexpr const _Wx& operator[](difference_type _Idx) const noexcept {
+                return *(*this + _Idx);
+            }
+
+            _NODISCARD_FRIEND constexpr bool operator==(const _Iterator& _Left, const _Iterator& _Right) noexcept(
+                noexcept(_Left._Current == _Right._Current)) /* strengthened */ {
+                return _Left._Current == _Right._Current;
+            }
+            _NODISCARD_FRIEND constexpr auto operator<=>(const _Iterator& _Left, const _Iterator& _Right) noexcept(
+                noexcept(_Left._Current <=> _Right._Current)) /* strengthened */ {
+                return _Left._Current <=> _Right._Current;
+            }
+
+            _NODISCARD_FRIEND constexpr _Iterator operator+(_Iterator _Iter, difference_type _Off) noexcept(
+                noexcept(_Iter += _Off)) /* strengthened */ {
+                return _Iter += _Off;
+            }
+            _NODISCARD_FRIEND constexpr _Iterator operator+(difference_type _Off, _Iterator _Iter) noexcept(
+                noexcept(_Iter += _Off)) /* strengthened */ {
+                return _Iter += _Off;
+            }
+
+            _NODISCARD_FRIEND constexpr _Iterator operator-(_Iterator _Iter, difference_type _Off) noexcept(
+                noexcept(_Iter -= _Off)) /* strengthened */ {
+                return _Iter -= _Off;
+            }
+            // clang-format off
+            _NODISCARD_FRIEND constexpr difference_type operator-(const _Iterator& _Left,const _Iterator& _Right)
+                noexcept(noexcept(_Left._Current - _Right._Current)) /* strengthened */ {

```

Comment:
This can be unconditionally `noexcept`.

Reply:
see https://github.com/microsoft/STL/pull/3142#discussion_r993687438: this is unfortunately not true (unless I'm missing something)

Reply:
> see [#3142 (comment)](https://github.com/microsoft/STL/pull/3142#discussion_r993687438): this is unfortunately not true (unless I'm missing something)

Per [iterator.concept.winc] p5 and p7, `operator-` between integer-class types has the same effects as `x - y`, where `x` and `y` are values of "a unique hypothetical extended integer type". So the `operator-` must be `noexcept` because `x - y` is non-throwing.